src/libponyc/ast/ast.c

```diff
@@ -1221,6 +1221,85 @@ void ast_free_unattached(ast_t* ast)
     ast_free(ast);
 }
 
+static void get_next_token_pos(ast_t* ast, size_t* line, size_t* pos)
+{
+  assert(ast != NULL);
+  
+  while (ast->child != NULL)
+  {
+    ast = ast->child;
+    while (ast->sibling != NULL)
+      ast = ast->sibling;
+  }
+
+  *line = token_line_number(ast->t);
+  *pos = token_line_position(ast->t);
+
+  token_id id = ast_id(ast);
+  if (id == TK_STRING || id == TK_ID)
+    *pos += token_string_len(ast->t);
```

Comment:
@kulibali - you could look at adding a `size_t len` member to `token_t` and keeping track of such things from the start.  That's a much more invasive change, though.

I think the right answer is probably to build the quoted parameter list from scratch.  That is, "paraphrase" it instead of a direct "quote".

We already have a mechanism for printing type names, so that combined with printing the names of the parameters and a few commas should get you close enough for these purposes.